### PR TITLE
Add classes used by JFreeChart to Index annotated JDK

### PR DESCRIPTION
**Pull request by kelloggm**
These annotations are useful for our case study of JFreeChart. I don't really think that the annotations on `java.util.Calendar` are complete, but the really correct ones require typetools#1723. In particular, `Calendar#get(int)` has an odd semantics: an integer is passed as a code, and it returns integers in different ranges depending on what the passed input is. So, for instance, if `Calendar.MONTH` is passed, the result is `@IntRange(from = 1, to = 12)`, but if `Calendar.DAY_OF_MONTH` is passed, the result is `@IntRange(from = 1, to = 31)`. A fully correct specification would be dependent on the value of the integer passed.

Nevertheless, I think this PR should be merged now, and we can update those annotations separately once we have the ability to write postcondition annotations that have arguments.

### CODE REVIEW

File: checker/jdk/index/src/java/awt/image/ColorModel.java [hunk 1]
```diff
@@ -0,0 +1,1973 @@
...
+     * @param components the array to receive the color and alpha
+     * components of the specified pixel
+     * @param offset the offset into the <code>components</code> array at
+     * which to start storing the color and alpha components
+     * @return an array containing the color and alpha components of the
+     * specified pixel starting at the specified offset.
+     * @throws UnsupportedOperationException if this
+     *          method is not supported by this <code>ColorModel</code>
+     */
+    public int @SameLen("#2") @PolyValue [] getComponents(int pixel, int @PolyValue [] components, int offset) {
```
Comment (panacekcz) [inline checker/jdk/index/src/java/awt/image/ColorModel.java]: Value annotations usually go to statically-executable.astub

File: checker/jdk/index/src/java/awt/image/ColorModel.java [hunk 2]
```diff
@@ -0,0 +1,1973 @@
...
+     * components of the specified pixel
+     * @param offset the index into the <code>components</code> array at
+     * which to begin storing the color and alpha components of the
+     * specified pixel
+     * @return an array containing the color and alpha components of the
+     * specified pixel starting at the specified offset.
+     * @throws UnsupportedOperationException if this
+     *          method is not supported by this <code>ColorModel</code>
+     */
+    public int @SameLen("#2") [] getComponents(Object pixel, int[] components, int offset) {
```
Comment (panacekcz) [inline checker/jdk/index/src/java/awt/image/ColorModel.java]: Why not `@PolyValue` like above? (Also methods below.)

File: checker/jdk/index/src/java/awt/image/ColorModel.java (hunk 1 above)
Comment (panacekcz) [reply checker/jdk/index/src/java/awt/image/ColorModel.java]: `offset` should be `@NonNegative`? (Also methods below.)

File: checker/jdk/index/src/java/awt/image/ColorModel.java [hunk 3]
```diff
@@ -0,0 +1,1973 @@
...
+     *  component values for this <code>ColorModel</code> are not
+     *  conveniently representable in the unnormalized form
+     * @throws ArrayIndexOutOfBoundsException if
+     *  the <code>components</code> array is not large enough to
+     *  hold all of the color and alpha components starting at
+     *  <code>offset</code>
+     * @throws UnsupportedOperationException if this
+     *  method is not supported by this <code>ColorModel</code>
+     */
+    public int getDataElement(int[] components, int offset) {
```
Comment (panacekcz) [inline checker/jdk/index/src/java/awt/image/ColorModel.java]: Could be `@IndexFor("#1") int offset` like below?

File: checker/jdk/index/src/java/awt/image/ColorModel.java [hunk 4]
```diff
@@ -0,0 +1,1973 @@
...
+                                                     normOffset, null, 0);
+        return getDataElement(components, 0);
+    }
+
+    /**
+     * Returns a data element array representation of a pixel in this
+     * <code>ColorModel</code>, given an array of normalized color/alpha
+     * components.  This array can then be passed to the
+     * <code>setDataElements</code> method of a <code>WritableRaster</code>
+     * object.  An <code>ArrayIndexOu
```
Comment (panacekcz) [inline checker/jdk/index/src/java/awt/image/ColorModel.java]: Should not be `@IndexFor("this")`, because it is an index to `normComponents`.

Comment (panacekcz) [inline checker/jdk/index/src/java/awt/image/ColorModel.java]: Shold be `@NonNegative`? (Also method below.)

File: checker/jdk/index/src/java/text/DateFormatSymbols.java [hunk 5]
```diff
@@ -0,0 +1,849 @@
...
+            dfsyms = provider.getInstance(locale);
+        }
+        return dfsyms;
+    }
+
+    /**
+     * Gets era strings. For example: "AD" and "BC".
+     * @return the era strings.
+     */
+    public String @ArrayLen(2) [] getEras() {
```
Comment (panacekcz) [inline checker/jdk/index/src/java/text/DateFormatSymbols.java]: Value annotations usually go to statically-executable.astub

File: checker/jdk/index/src/java/util/Calendar.java [hunk 6]
```diff
@@ -0,0 +1,2769 @@
...
+     * system dependent.
+     *
+     * @param field the given calendar field.
+     * @return the value for the given calendar field.
+     * @throws ArrayIndexOutOfBoundsException if the specified field is out of range
+     *                                        (<code>field &lt; 0 || field &gt;= FIELD_COUNT</code>).
+     * @see #set(int, int)
+     * @see #complete()
+     */
+    public int get(int field) {
```
Comment (panacekcz) [inline checker/jdk/index/src/java/util/Calendar.java]: Could be `@NonNegative int field`. Also other public methods below.

File: checker/jdk/index/src/java/util/Calendar.java [hunk 7]
```diff
@@ -0,0 +1,2769 @@
...
+    }
+
+    /**
+     * Returns whether the value of the specified calendar field has been set
+     * externally by calling one of the setter methods rather than by the
+     * internal time calculation.
+     *
+     * @return <code>true</code> if the field has been set externally,
+     * <code>false</code> otherwise.
+     * @throws IndexOutOfBoundsEx
```
Comment (panacekcz) [inline checker/jdk/index/src/java/util/Calendar.java]: Could be `@IntRange(from = 1, to = 7) int value`. Also `setMinimalDaysInFirstWeek` below.

**Comment by kelloggm** [comment]
Thanks for your comments. I believe I've addressed them all.

File: framework/src/org/checkerframework/common/value/statically-executable.astub [hunk 8]
```diff
@@ -360,3 +360,39 @@ class Arrays {
     static @MinLen(2) String toString(Object[] a);
     static @MinLen(2) String deepToString(Object[] a);
 }
+
+class Calendar {
+    public @IntRange(from = 1, to = 7) int getFirstDayOfWeek();
+    public void setFirstDayOfWeek(@IntRange(from = 1, to = 7) int value);
+    public @IntRange(from = 1, to = 7) int getMinimalDaysInFirstWeek();
+    public void setMinimalDaysInFirstWeek(@IntRange(from = 1, to = 7) int value) {
```
Comment (panacekcz) [inline framework/src/org/checkerframework/common/value/statically-executable.astub]: Declaration should end with a `;`.

Comment (kelloggm) [reply framework/src/org/checkerframework/common/value/statically-executable.astub]: Sorry you had to find this - this was just carelessness on my part. It should be fixed now.

**Comment by mernst** [comment]
@kelloggm The merge conflict is the git hash for the downloaded `jdk8.jar`.  Could you please update so that we can get this pull request merged?  Thanks!

**Comment by kelloggm** [comment]
Updated to a new JDK. I now expect tests to pass. The issue was that other changes to the JDK had been merged since this PR was opened.

**Pull request merged**